radix-engine-tests/tests/blueprints/account_locker.rs

```diff
@@ -1418,32 +1418,28 @@ fn state_of_the_account_locker_can_be_reconciled_from_events_alone() {
             )
         });
 
+    fn set_disallowed_preference(builder: ManifestBuilder, account: ComponentAddress, resource: ResourceAddress) -> ManifestBuilder {
+        builder.call_method(
+            account,
+            ACCOUNT_SET_RESOURCE_PREFERENCE_IDENT,
+            AccountSetResourcePreferenceInput {
+                resource_address: resource,
+                resource_preference: ResourcePreference::Disallowed,
+            }
+        )
+    }
+
+    let builder = ManifestBuilder::new();
```

Comment:
This is test code so this is just a recommendation, but I really like the use for things like this and I think that we can use `fold` to make this quite neat. Something like the following achieves the same effect:

```rust

let manifest = [
    (user_account1, fungible_resource1),
    (user_account1, non_fungible_resource3),
    (user_account2, fungible_resource2),
    (user_account2, non_fungible_resource2),
    (user_account3, fungible_resource3),
    (user_account3, non_fungible_resource1),
]
.iter()
.fold(ManifestBuilder::new(), |builder, (account, resource)| {
    builder.call_method(
        account,
        ACCOUNT_SET_RESOURCE_PREFERENCE_IDENT,
        AccountSetResourcePreferenceInput {
            resource_address: resource,
            resource_preference: ResourcePreference::Disallowed,
        },
    )
})
.build();
```

Reply:
I have found myself using the above fold pattern with the manifest builder very very often lately and it very quickly grew on me.

Reply:
This is a nice idea in principle. That said, I have a semi-irrational dislike for `fold` / `reduce` - basically these guys capture my feelings well: https://www.youtube.com/watch?v=qaGjS7-qWzg

Typically using a loop (or explicit unrolling in this case) is simpler / easier to follow in my opinion. :)